prow/cmd/deck/rerun.go

```diff
@@ -105,7 +244,39 @@ func handleRerun(cfg config.Getter, prowJobClient prowv1.ProwJobInterface, creat
 			}
 			return
 		}
-		newPJ := pjutil.NewProwJob(pj.Spec, pj.ObjectMeta.Labels, pj.ObjectMeta.Annotations)
+		var newPJ v1.ProwJob
+		if mode == LATEST {
+			prowJobSpec, labels, annotations, err := getProwJobSpec(pj.Spec.Type, cfg, pj.Spec.Job, pj.Spec.Refs, pj.Labels)
+			if err != nil {
+				// These are user errors, i.e. missing fields, requested prowjob doesn't exist etc.
+				// These errors are already surfaced to user via pubsub two lines below.
+				http.Error(w, "Could not create new prowjob: Failed getting prowjob spec", http.StatusInternalServerError)
+				l.WithError(err).Error("Could not create new prowjob")
```

Comment:
Sure this should be on error level? This is a mistake by the user and not an error in deck or not?

Reply:
Agreed, this has been changed from `http.StatusInternalServerError` to `http.StatusBadRequest`

Reply:
```suggestion
				l.WithError(err).Debug("Could not create new prowjob")
```

Reply:
Updated!